### PR TITLE
Fix secondary rate limits URL and improve related documentation

### PR DESCRIPTION
**Pull request by rpelliard**
It looks like GitHub is now designating "abuse rate limits" as "secondary rate limits": https://docs.github.com/en/rest/overview/resources-in-the-rest-api#secondary-rate-limits

The URL returned in the error response also changed, so this PR updates it so that these errors can properly be caught again.
This PR also includes a suggestion to update the main documentation page to mention this error type, to make it more discoverable.

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @rpelliard !
LGTM.

Awaiting second LGTM before merging.
Any other repo contributor can approve for the second LGTM.

**Review by pierce-m — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @pierce-m !
Merging.

**Pull request merged**
